IPython/core/magics/packaging.py

```diff
@@ -10,12 +10,23 @@
 
 import re
 import shlex
+import subprocess
 import sys
 from pathlib import Path
 
 from IPython.core.magic import Magics, magics_class, line_magic
 
 
+def _format_command(*args):
+    """Use subprocess.list2cmdline to quote and escape executable path"""
+    return subprocess.list2cmdline(list(args))
+
+
+def _get_full_path(path):
+    """Return a normalised path of the current Python executable"""
+    return Path(sys.executable).expanduser().absolute().resolve()
```

Comment:
This is unnecessary. Judging by [the docs](https://docs.python.org/3/library/sys.html#sys.executable), `sys.executable` is already an absolute path, so `expanduser().absolute().resolve()` will have no effect on it.

I don't have macOS (vide the issue), so I cannot check it. Could you please explain this, and maybe show the values of `sys.executable` this is meant to shield against?

Reply:
You're right about `sys.executable`.
I used this function for testing purposes, with different paths, but now it seems redundant.
Beside of that, I forgot to handle the `path` argument.